source/extensions/filters/http/json_to_metadata/filter.h

```diff
@@ -56,14 +56,14 @@ using Rules = std::vector<Rule>;
 /**
  * Configuration for the Json to Metadata filter.
  */
-class FilterConfig {
+class FilterConfig : public ::Envoy::Router::RouteSpecificFilterConfig {
 public:
-  FilterConfig(
-      const envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata& proto_config,
-      Stats::Scope& scope, Regex::Engine& regex_engine);
+  static absl::StatusOr<std::shared_ptr<FilterConfig>>
+  create(const envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata& proto_config,
+         Stats::Scope& scope, Regex::Engine& regex_engine, bool per_route = false);
 
-  JsonToMetadataStats& rqstats() { return rqstats_; }
-  JsonToMetadataStats& respstats() { return respstats_; }
+  JsonToMetadataStats& rqstats() const { return rqstats_; }
+  JsonToMetadataStats& respstats() const { return respstats_; }
```

Comment:
I don't think you should mark this const if we're planning to mutate the state outside of it.

e.g. see  "logical constness" versus "bitwise constness"